models/ApiUser.php

```diff
@@ -58,6 +58,7 @@ public function load($data, $formName = null)
         $result = parent::load($data, $formName);
         $this->user->authclient_id = $this->authclient_id;
 
+        $this->user->scenario = User::SCENARIO_EDIT_ADMIN;
```

Comment:
@yurabakhtin I cannot remove `$this->user->scenario = User::SCENARIO_EDIT_ADMIN;` otherwise visibility value is not loaded to the model because the rules in the User model about the visibility property is set only with this scenario: https://github.com/humhub/humhub/blob/master/protected/humhub/modules/user/models/User.php#L163

But yes, you're right, we need to check if the user is admin for these fields. So I've added 2 `if (Yii::$app->user->isAdmin()) {` before setting the `SCENARIO_EDIT_ADMIN` to the User and Profil models, and I've moved this scenario setting to the controller. See commit https://github.com/humhub/rest/pull/140/commits/ba0257bfe97e89d7a7968d48db4aedc0c0e39523

However, according to https://github.com/humhub/humhub/blob/master/protected/humhub/modules/admin/controllers/UserController.php#L166-L178 , we should also restrict the `status` property for admins. But the the `status` property is not restricted to admins in the rules: https://github.com/humhub/humhub/blob/master/protected/humhub/modules/user/models/User.php#L162

Perhaps we should add `, 'on' => static::SCENARIO_EDIT_ADMIN` in the rules?

Reply:
@marc-farre Yes it seems you are right, thank you for the note.
We have to check how it is possible to update the field `status` when current user has no permission.

But why did you use only one condition `Yii::$app->user->isAdmin()`?
I think we should use there the same condition what we have in backoffice part:

- `Yii::$app->user->isAdmin() || !$apiUser->user->isSystemAdmin()`.

I.e. if the updating user is not Admin then current user manager can update the columns `status` and `visibility`.

Reply:
Yes, sorry. Fixed in commit https://github.com/humhub/rest/pull/140/commits/0201c70fbfdfd921b0ebb4ed57f692a7e7520a6c

Reply:
@marc-farre PR https://github.com/humhub/humhub/pull/6578 for your remarks from the comment https://github.com/humhub/rest/pull/140#discussion_r1330402345.

Reply:
@yurabakhtin Should I merge #6578 into `develop` (v1.15)  or is `v1.16` also fine?

Reply:
@luke- the #6578 should be merged into `master` (v1.14) because a vulnerability was fixed there.